### PR TITLE
repo

### PR DESCRIPTION
**Pull request by valeIT**

### CODE REVIEW

File: _posts/2019-11-09-view-model-repository.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+         guard let cell = tableView.dequeueReusableCell(withIdentifier: cellReuseIdentifier, for: indexPath) as? UserTableViewCell else {
+            fatalError()
+        }
+        let model = model[indexPath.row]
+        cell.populate(withModel: model)
+        return cell
+    }
+```
+
+Every call to cellForRowAt indexPath should be this short. Configuring the cell should be done from the cell itself and not in the controller. Look for opportunities to move logic where it should be. When starting out with MVC we find ourselves to put everything in the controller without thinking where the right place for this function should be.
```
Comment (MsSlacker) [inline _posts/2019-11-09-view-model-repository.md]: What kind of sentence is "we find ourselves to put...". It sounds shit.

File: _posts/2019-11-09-view-model-repository.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+extension AmountViewModel {
+    init(model: AmountModel, formatter: NumberFormatter) {//1
+        self.amount = formatter.string(from: model.amount as NSNumber) ?? ""
+    }
+}
+```
+
+1. Note how the view model doesn't know about the formatter, it gets told what formatter to use.
+
+Usually people using MVVM put all the logic inside the viewModel. I strongly disagree. The viewModel should be dumb, as the model, the should only know how to configure themselves. They can have helper methods, but they should not communicate directly to databases or initiate networking calls.
```
Comment (MsSlacker) [inline _posts/2019-11-09-view-model-repository.md]: the - > it
themselves - > itself

Comment (MsSlacker) [reply _posts/2019-11-09-view-model-repository.md]: Unless you by it to stay plural, but still the - > they

Comment (MsSlacker) [reply _posts/2019-11-09-view-model-repository.md]: Which means there's 3 they in there...

File: _posts/2019-11-09-view-model-repository.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        fatalError()
+    }
+    let model = repository.viewModel[indexPath.row]
+    //Here we access the view model, we could also expose it using a method or a protocol if we would want to enforce more separation of concerns, but this one is a great first step.
+    cell.populate(withModel: model)
+    return cell
+}
+```
+
+Note how the controller holds a strong reference to the repository. This way once the controller goes out of scope the repository will get deallocated as well.
```
Comment (MsSlacker) [inline _posts/2019-11-09-view-model-repository.md]: de-allocated? Looks very funny all together like this...

**Review by MsSlacker — CHANGES_REQUESTED** [review]

File: _posts/2019-11-09-view-model-repository.md (hunk 3 above)
Comment (valeIT) [reply _posts/2019-11-09-view-model-repository.md]: mm nope https://en.wiktionary.org/wiki/deallocate

File: _posts/2019-11-09-view-model-repository.md (hunk 2 above)
Comment (valeIT) [reply _posts/2019-11-09-view-model-repository.md]: mmm not so sure about this.
model and view model are 2 -> plural

**Review by MsSlacker — APPROVED** [review]

**Pull request merged**